metrix-mapping/src/main/java/com/powsybl/metrix/mapping/timeseries/FileSystemTimeSeriesStore.java

```diff
@@ -316,6 +329,42 @@ private TimeSeries appendTimeSeries(TimeSeries existingTimeSeries, TimeSeries ne
         }
     }
 
+    private Set<Integer> getChunkIndexes(TimeSeries<?, ?> timeSeries, TimeSeriesDataType dataType) {
+        Set<Integer> existingChunkIndexes = new HashSet<>();
+        if (dataType.equals(TimeSeriesDataType.DOUBLE)) {
+            existingChunkIndexes.addAll(((StoredDoubleTimeSeries) timeSeries).getChunks().stream()
+                .flatMap(chunk -> IntStream.range(chunk.getOffset(), chunk.getOffset() + chunk.getLength()).boxed())
+                .collect(Collectors.toSet()));
+        } else {
+            existingChunkIndexes.addAll(((StringTimeSeries) timeSeries).getChunks().stream()
+                .flatMap(chunk -> IntStream.range(chunk.getOffset(), chunk.getOffset() + chunk.getLength()).boxed())
+                .collect(Collectors.toSet()));
+        }
+        return existingChunkIndexes;
+    }
+
+    private TimeSeries appendTimeSeriesWithSameIndex(TimeSeries<?, ?> existingTimeSeries, TimeSeries<?, ?> newTimeSeries, TimeSeriesDataType dataType) {
+        // Check that the timeseries don't have chunks with the same offset
+        Set<Integer> existingOffsets = getChunkIndexes(existingTimeSeries, dataType);
```

Comment:
offset in variable name and function name is confusing here -> point would be better 
existingOffsets -> existingPoints
newOffsets -> newPoints
getChunkIndexes -> getChunkPoints